pkg/k8shandler/visualization.go

```diff
@@ -236,7 +236,14 @@ func (clusterRequest *ClusterLoggingRequest) createOrUpdateKibanaDeployment(prox
 		return
 	}
 
-	kibanaPodSpec := newKibanaPodSpec(clusterRequest.cluster, "kibana", "elasticsearch.openshift-logging.svc.cluster.local", proxyConfig, kibanaTrustBundle)
+	routerCanonicalHostname, err := clusterRequest.GetRouterCanonicalHostname("kibana")
+	if err != nil {
+		logrus.Error("Unable to get routerCanonicalHostname for kibana")
+	}
+
+	oauthIssuer := fmt.Sprintf("%s.%s", "oauth-openshift", routerCanonicalHostname)
```

Comment:
@bparees is it appropriate to always assume the oauth issuer will be "oauth-openshift" ? Are users able to configure this at all?

Reply:
yes, i would expect even if the cluster has an alternate IDP configured, you're still talking to openshift-oauth and i'd be surprised if the value is cluster-admin configurable.  But @deads2k or someone from the apiserver team would be best to consult with.

Reply:
Keycloak will replace oauth-server by using their own authn webhook. I miss the context here to tell whether this matters for the question. Maybe @stlaz has more insights.

Reply:
Like David said, this approach should not be used.

Also, you should by no means assume the issuer of the user-tokens uses the oauth-openshift route, we are working on abstracting the token validation to any service capable of dealing with kubernetes webhook authentication flow, meaning users will be able to configure the oauth-server to whatever they please.